src/Sylius/Bundle/ShopBundle/templates/shared/layout/base/header/top_bar/locale_switch.html.twig

```diff
@@ -1,14 +1,23 @@
 {% if available_locales|length > 1 %}
     <div class="col-auto">
-        <div {{ sylius_test_html_attribute('locale-selector') }}>
-            <span {{ sylius_test_html_attribute('active-locale') }}>{{ active_locale|sylius_locale_name }}</span>
-            {% for code in available_locales %}
-                {% if code is not same as(active_locale) %}
-                    <a href="{{ path('sylius_shop_switch_locale', {'code': code}) }}" {{ sylius_test_html_attribute('available-locale') }}>
-                        {{ code|sylius_locale_name }}
-                    </a>
-                {% endif %}
-            {% endfor %}
+        <div class="dropdown" {{ sylius_test_html_attribute('locale-selector') }}>
+            <button class="btn btn-sm btn-dark dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" {{ sylius_test_html_attribute('active-locale', active_locale) }}>
+                <span class="fi fi-{{ active_locale|sylius_locale_country|lower }} me-2"></span>
+                {{ active_locale|sylius_locale_country }}
+            </button>
+                {{ dump(available_locales) }}
+            <ul class="dropdown-menu dropdown-menu-end dropdown-menu-dark">
+                {% for code in available_locales %}
+                    {% if code is not same as(active_locale) %}
+                        <li>
+                            <a class="dropdown-item py-2" href="{{ path('sylius_shop_switch_locale', {'code': code}) }}" {{ sylius_test_html_attribute('available-locale', code) }} >
+                                <span class="fi fi-{{ code|sylius_locale_country|lower }} me-2"></span>
+                                {{ code|sylius_locale_country }}
```

Comment:
```suggestion
                                {% set country_code = code|sylius_locale_country %}
                                <span class="fi fi-{{ country_code|lower }} me-2"></span>
                                {{ country_code }}
```
Though the comment from the active locale is also applicable in this case

Reply:
same here

Reply:
But the above suggestion with setting a variable could be applied